crates/ruff_linter/resources/test/fixtures/flake8_type_checking/TC006.py

```diff
@@ -0,0 +1,62 @@
+def f():
+    from typing import cast
+
+    cast(int, 3.0)  # TC006
+
+
+def f():
+    from typing import cast
+
+    cast(list[tuple[bool | float | int | str]], 3.0)  # TC006
+
+
+def f():
+    from typing import Union, cast
+
+    cast(list[tuple[Union[bool, float, int, str]]], 3.0)  # TC006
+
+
+def f():
+    from typing import cast
+
+    cast("int", 3.0)  # OK
+
+
+def f():
+    from typing import cast
+
+    cast("list[tuple[bool | float | int | str]]", 3.0)  # OK
+
+
+def f():
+    from typing import Union, cast
+
+    cast("list[tuple[Union[bool, float, int, str]]]", 3.0)  # OK
+
+
+def f():
+    from typing import cast as typecast
+
+    typecast(int, 3.0)  # TC006
+
+
+def f():
+    import typing
+
+    typing.cast(int, 3.0)  # TC006
+
+
+def f():
+    import typing as t
+
+    t.cast(int, 3.0)  # TC006
```

Comment:
Could we add a test where parts of the type are quoted?
```suggestion
    t.cast(Literal["3"] | Literal['3'], 3.0)  # TC006
```